### PR TITLE
Add option to change disposal title via sign

### LINKED ISSUE #2626: 1.14.3 - all commands broken
## Information

Full output of `/ess version`: _All commands including `/ess version` are broken, cannot get output_
EssentialsX file name: EssentialsX-2.16.1.215.jar
EssentialsX version: 2.16.1.215

[Server log & EssentialsX config](https://gist.github.com/sportshead/efabc9b3ee183001a4d2d7afe6b50ed0)

### Problem
When trying to execute commands, console returns first while typing command: `An internal error occurred while attempting to tab-complete this command`
After hitting <kbd>enter</kbd>, console returns `An internal error occurred while attempting to perform this command`

### What I have tried

- [ ] Restarting server -- fail

### Screenshots

[Video of attempted `/gamemode creative`](https://streamable.com/jz3io)

### LINKED ISSUE #2628: Allow changing the inv title of [disposal] through the sign
### Feature request

**Feature description**

Allow players (that perhaps also have permission for it like `essentials.disposal.rename`) to rename the inventory-title of the `[Disposal]` sign by adding the name to the lines in the sign (See image for example).

**How the feature is useful**

PR #2461 added an option for server owners (or devs) to rename the default inventory-title through the properties file of essentials.
However this is limited to the just mentioned owner or devs and is only one title.
Perhaps some players want to have an own (funny) title to show instead of the "Disposal" title.
Something like "Black hole".

**Images**
Here is an image that would show how it looks like on a sign.
First line is the default `[disposal]` text while second (And maybe 3rd and 4th) line is for the name.
image

----
Another Idea that just came to mind is to add a payment option for using disposal (if that isn't already a thing.)
When a player would click on a disposal sign (that perhaps has the price set on the 4th line) EssentialsX would tell them that they have to pay X mones to use it. The player could then rightclick again to confirm it or let it time out.

### PR DESCRIPTION
**Pull request by JRoy**
wHy iS tHiS eSsEnTaIl

Closes #2628

### CODE REVIEW

**Review by Ichbinjoe — DISMISSED** [review]
Should we allow this ungated behind a permission node (to place the signs)? Also are there any considerations for backwards compat. if we were to go down this permission model? I realize this isn't really a problem in MC but this makes me feel like this is something a user could use to trick another player (I mean its Minecraft but people are assholes)

**Comment by JRoy** [comment]
Players can write things on the disposal signs anyways, this just translates that onto the inventory title. Not worth a permission node + backward compat imo @Ichbinjoe

**Comment by pop4959** [comment]
How does it look on a sign to trigger this feature vs old versions of the sign?

**Comment by JRoy** [comment]
@pop4959 
You could never change the title before but now you can write on lines 2-4 to change the title of the sign when a user has a certain permission. If the user lacks said permission, the rest of the lines 2-4 on the sign wipe so the title cannot be changed.

File: Essentials/src/com/earth2me/essentials/signs/SignDisposal.java [hunk 1]
```diff
@@ -11,9 +12,23 @@ public SignDisposal() {
...
+    }
+
     @Override
     protected boolean onSignInteract(final ISign sign, final User player, final String username, final IEssentials ess) {
-        player.getBase().openInventory(ess.getServer().createInventory(player.getBase(), 36, tl("disposal")));
+        String title = sign.getLine(1) + " " + sign.getLine(2) + " " + sign.getLine(3);
+        if (title.trim().isEmpty()) {
+            title = tl("disposal");
+        }
+        player.getBase().openInventory(ess.getServer().createInventory(player.getBase(), 36, title.trim()));
```
Comment (mdcfe) [inline Essentials/src/com/earth2me/essentials/signs/SignDisposal.java]: This is a *very* subtle breaking change, but if server admins intentionally add leading spaces to the `disposal` locale string, `title.trim()` will remove them and break the layout. I'd prefer trimming the title only if it was set on the sign. Granted, I'm not sure if anyone really does this, but on the off chance people do we'll be breaking their disposal signs.

Comment (JRoy) [reply Essentials/src/com/earth2me/essentials/signs/SignDisposal.java]: Changes applied :P

**Review by mdcfe — APPROVED** [review]

**Review by darbyjack — APPROVED** [review]

**Review by mbax — APPROVED** [review]

**Pull request merged**